src/error.rs

```diff
@@ -41,6 +40,13 @@ pub enum Error {
     Io(IoError),
     /// An internal error occurred while decoding the image.
     Internal(Box<dyn StdError + Send + Sync + 'static>), //TODO: not used, can be removed with the next version bump
+    /// An error that occurred during the decode, but allows for incomplete image pixel data to be returned with `try_recover()`
+    Recoverable {
+        /// Error occurred while decoding the image
+        err: Box<Self>,
+        /// Incomplete pixel data of the image
+        pixels: Vec<u8>
+    }
```

Comment:
If the variant stored a `struct RecoverableError` that contains the fields instead we could later add different error information and state in a SemVer compatible upgrade. And provide methods specifically for recoverable errors, such as listing of errors or a summary of their kinds / specificity. Nothing specific comes to mind but it's been quite useful to keep the option available.

Compare roughly design of `image::ImageError` for *some inspiration*, maybe, but I don't want to cause rash complexification either. Simply these fields in a struct seems fine to me for the moment. That would answer the question of multiple errors by not having to answer it, yet :slightly_smiling_face: